### PR TITLE
scale up the work agent deployment after the hub-kuebeconfig-secret is created

### PR DESCRIPTION
**Pull request by zhiweiyin318**
fix issue https://github.com/open-cluster-management-io/community/issues/111

scale up the work agent deployment when the hubConnectionDegraded condition of klusterlet is False. 

Signed-off-by: Zhiwei Yin <zyin@redhat.com>

### CODE REVIEW

File: pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go [hunk 1]
```diff
@@ -335,15 +330,22 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
...
 	relatedResources = append(relatedResources, statuses...)
 
-	// If cluster name is empty, read cluster name from hub config secret
+	// If cluster name is empty, read cluster name from hub config secret.
+	// registration-agent generated the cluster name and set it into hub config secret.
 	if config.ClusterName == "" {
-		clusterName := hubSecret.Data["cluster-name"]
-		if clusterName != nil {
-			config.ClusterName = string(clusterName)
+		config.ClusterName, err = n.getClusterNameFromHubKubeConfigSecret(ctx, &config)
```
Comment (qiujian16) [inline pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go]: if using pointer here. it does not need to return and set, right?

Comment (zhiweiyin318) [reply pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go]: no, I don't think we should deploy work agent deployment if clusterName is empty when failed to get hub-kubeconfig secret or there is no clusterName in hub-kubeconfig secret.

Comment (zhiweiyin318) [reply pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go]: I got your point.  you mean this ?
```
clusterName := n.getClusterNameFromHubKubeConfigSecret(ctx, &config)
if clusterName == nil {
 return fmt.Errorf("failed to get clusterName")
}
```

Comment (qiujian16) [reply pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go]:
```go
err := n.setClusterNameFromHubKubeConfigSecret(ctx, &config)
if err != nil {
  return err
}
```

Comment (zhiweiyin318) [reply pkg/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go]: done

**Review by yue9944882 — COMMENTED** [review]
LGTM

**Pull request merged**